.circleci/config.yml

```diff
@@ -55,8 +73,25 @@ jobs:
           #   The test command
           test-command: make ci-functional-test
       
+      # code coverage
+      - run:
+          name: Upload Code Coverage Report
+          command: |
+            curl -s https://codecov.io/bash > codecov;
+            VERSION=$(grep 'VERSION=\"[0-9\.]*\"' codecov | cut -d'"' -f2);
+            SHAVERSION=$(shasum -v);
+            echo "Using CodeCov version '$VERSION'"
+            echo "Using shasum '$SHAVERSION'"
+            for i in 1 256 512
+            do
+              shasum -a $i -c --ignore-missing <(curl -s "https://raw.githubusercontent.com/codecov/codecov-bash/${VERSION}/SHA${i}SUM") ||
+              shasum -a $i -c <(curl -s "https://raw.githubusercontent.com/codecov/codecov-bash/${VERSION}/SHA${i}SUM" | grep -w "codecov")
+            done
```

Comment:
Can we have this as a separate step - Download codecov - and then run the corresponding command in each of these steps? Core uses something like that if you want to check it out.